modules/debug_cmd/debug_cmd.c

```diff
@@ -143,30 +143,34 @@ static int cmd_play_file(struct re_printf *pf, void *arg)
 {
 	struct cmd_arg *carg = arg;
 	struct config *cfg;
-	const char *filename = carg->prm;
+	char filename[4096] = { 0 };
+	int offset = 0;
 	int err = 0;
 
-	cfg = conf_config();
-
 	/* Stop the current tone, if any */
 	g_play = mem_deref(g_play);
 
-	if (str_isset(filename))
-	{
-		err = re_hprintf(pf, "playing audio file \"%s\" ..\n",
-				 filename);
+	if (str_isset(carg->prm) && sscanf(carg->prm, "%4095s %d", filename, &offset)) {
```

Comment:
use `re_regex()` instead of `sscanf()`

With `pl_strdup()` you get a C string from a `pl`

Then there is no need for a large buffer.